src/wp-includes/class-wp-block.php

```diff
@@ -223,7 +223,7 @@ public function render( $options = array() ) {
 			foreach ( $this->inner_content as $chunk ) {
 				if ( is_string( $chunk ) ) {
 					$block_content .= $chunk;
-				} else {
+				} else if ( ! empty( $this->inner_blocks[ $index ] ) ) {
```

Comment:
I think a safer change is a check for specifically is_null.

For context:
`$this->inner_blocks` is assigned a new WP_Block_List; the offsetGet method (i.e. `$this->inner_blocks[ $index ]`) can return [either a block value, or null](https://opengrok.a8c.com/source/xref/wpcom/wp-includes/class-wp-block-list.php?r=36124d3f&fi=WP_Block_List#89).

In PHP 8, trying to assign a property to null was escalated from a **warning** (Creating default object from empty value) to a **fatal** (Fatal error: Uncaught Error: Attempt to assign property "bar" on null).

I don't know under what circumstances an empty WP_Block_List object might be returned, but given how integral this function is to blocks working, I'd keep changes as small as possible.